citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java

```diff
@@ -43,8 +47,24 @@ public String showPage(Authentication authentication, Model model, CitationForm
         if (!model.containsAttribute("form")) {
             model.addAttribute("form", form);
         }
+        List<String> fields = new ArrayList<>();
+        citationManager.getItemTypeFields((IUser) authentication.getPrincipal(), citation.getItemType())
+            .forEach(f -> fields.add(f.getFilename()));
+        model.addAttribute("fields", fields);
         return "auth/group/items/item/edit";
     }
+
+    
+    /**
+     * Method to retrieve all fields filtered by item type.
+     */
+    @RequestMapping("/auth/items/fields")
+    public ResponseEntity<List<String>> getFieldsByItemType(Authentication authentication, @RequestParam(value = "itemType") ItemType itemType) {
```

Comment:
I would make the itemType request parameter a path variable (to make it a little bit more rest-like)